test/Confluent.Kafka.IntegrationTests/Tests/AddBroker.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Text;
+using System.Collections.Generic;
+using Confluent.Kafka.Serialization;
+using Xunit;
+
+
+namespace Confluent.Kafka.IntegrationTests
+{
+    public static partial class Tests
+    {
+        /// <summary>
+        ///     Test that produces a message then consumes it.
+        /// </summary>
+        [Theory, MemberData(nameof(KafkaParameters))]
+
+        public static void AddBrokers(string bootstrapServers, string topic, string partitionedTopic)
+        {
+            // This test assumes broker v0.10.0 or higher:
+            // https://github.com/edenhill/librdkafka/wiki/Broker-version-compatibility
+
+            // This test call metadata, as it's an easy way to see if we are connected to broker
+            // It's not really the best way to test it (ideally, we would get from a working list of brokers
+            // to all brokers have change IP) but this wil be good enough
+
+            var producerConfig = new Dictionary<string, object>
+            {
+                { "bootstrap.servers", "unknown" },
+                { "api.version.request", true }
+            };
+
+            var consumerConfig = new Dictionary<string, object>
+            {
+                { "group.id", Guid.NewGuid().ToString() },
+                { "bootstrap.servers", "unknown" },
+                { "session.timeout.ms", 6000 },
+                { "api.version.request", true }
+            };
+
+            using (var typedProducer = new Producer<Null, string>(producerConfig, null, new StringSerializer(Encoding.UTF8)))
+            {
+                TestMetadata(
+                    () => typedProducer.GetMetadata(false, null, TimeSpan.FromSeconds(3)),
```

Comment:
Would take advise on how we should check if brokers are connected or not. This seemed a good way, perhaps there are better.